### PR TITLE
Fix metadata tracker re-scheduling/operation chaining 

### PR DESCRIPTION
**Pull request by seut**
Use completion future chain for all operations:
 1. process changed & dropped relations/partitions
 2. process removed relations/partitions.

Only release and re-schedule after both steps are done to prevent that a new schedule runs concurrently with a previous one.

2nd commit fixes an NPE inside the `RestoreService` if the repositories metadata changed in-between API calls.

### CODE REVIEW

File: server/src/main/java/io/crate/replication/logical/MetadataTracker.java [hunk 1]
```diff
@@ -223,7 +222,41 @@ private void run() {
                 continue;
             }
 
-            ActionListener<AcknowledgedResponse> listener = ActionListener.wrap(r -> countDown.onSuccess(), onError);
+            var clusterStateListener = new FutureActionListener<>(AcknowledgedResponse::isAcknowledged);
+            clusterStateListener
+                .whenComplete(
+                    (r, e) -> {
+                        // Only check for new relations and partitions if deleted relations and partitions are processed
+                        var stateFuture = getPublicationsState(subscription, client);
```
Comment (mfussenegger) [inline server/src/main/java/io/crate/replication/logical/MetadataTracker.java:230]: Is this intentionally not composing? `clusterStateListener.whenComplete` will complete without waiting for the publicationsState future

Comment (seut) [reply server/src/main/java/io/crate/replication/logical/MetadataTracker.java:230]: no, thx for catching!

**Review by mfussenegger — APPROVED** [review]

**Comment by seut** [comment]
retest this please

**Comment by seut** [comment]
retest this please

**Pull request merged**